### PR TITLE
Deprecate FindBugs plugin

### LINKED ISSUE #6652: Deprecate JDepend and FindBugs plugins
Since both projects are unmaintained and don't work with Java 9/10/11 bytecode, the corresponding plugins should be deprecated.

For FindBugs, we could point users to [SpotBugs](https://spotbugs.github.io/) which provides a community Gradle plugin. For JDepend there's no clear successor.

### PR DESCRIPTION
**Pull request by marcphilipp**
Issue: #6652.

### CODE REVIEW

File: subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/FindBugsPlugin.java [hunk 1]
```diff
@@ -61,6 +64,7 @@ protected String getToolName() {
 
     @Override
     protected void beforeApply() {
+        SingleMessageLogger.nagUserOfPluginReplacedWithExternalOne("FindBugs", "SpotBugs");
```
Comment (oehme) [inline subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/FindBugsPlugin.java]: ❌ Should use the id's of the plugins so users can find them easier.

Comment (marcphilipp) [reply subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/FindBugsPlugin.java]: Same for JDepend, I assume, right?

Comment (oehme) [reply subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/FindBugsPlugin.java]: Right.

**Comment by marcphilipp** [comment]
I documented the deprecations in the plugins' chapters in the user guide.

@oehme PTAL!

**Pull request merged**